Src/Private/Invoke-LabModuleDownloadFromAzDo.ps1

```diff
@@ -0,0 +1,67 @@
+function Invoke-LabModuleDownloadFromAzDo {
+    <#
+    .SYNOPSIS
+        Downloads a PowerShell module/DSC resource from an Azure DevOps Feed to the host's module cache.
+#>
+    [CmdletBinding(DefaultParameterSetName = 'Latest')]
+    [OutputType([System.IO.FileInfo])]
+    param (
+        ## PowerShell module/DSC resource module name
+        [Parameter(Mandatory, ValueFromPipeline, ValueFromPipelineByPropertyName)]
+        [System.String] $Name,
+
+        ## Destination directory path to download the PowerShell module/DSC resource module to
+        [Parameter(ValueFromPipeline, ValueFromPipelineByPropertyName)]
+        [ValidateNotNullOrEmpty()]
+        [System.String] $DestinationPath = (Get-ConfigurationData -Configuration Host).ModuleCachePath,
+
+        ## The minimum version of the module required
+        [Parameter(Mandatory, ValueFromPipelineByPropertyName, ParameterSetName = 'MinimumVersion')]
+        [System.Version] $MinimumVersion,
+
+        ## The exact version of the module required
+        [Parameter(Mandatory, ValueFromPipelineByPropertyName, ParameterSetName = 'RequiredVersion')]
+        [System.Version] $RequiredVersion,
+
+        # if I uncomment this I get the error that the parameter is not recognized
+        # but if I comment it out, the script works !!
+        #[Parameter(ValueFromPipelineByPropertyName)]
+        #[AllowNull()]
+        #[System.Management.Automation.PSCredential]
+        #[System.Management.Automation.CredentialAttribute()]
+        #$FeedCredential,
```

Comment:
I can't see anything obvious here, but it obviously needs resolving. It'll just be grabbing the $FeedCredential from a parent scope otherwise.

Reply:
I did some digging and worked out what's going on here. The `Resolve-AzDoModuleUri` and `Invoke-LabModuleDownloadFromAzDo` functions have the `ValueFromRemainingArguments` catch-all defined. When `Invoke-LabModuleDownloadFromAzDo` calls `Resolve-AzDoModuleUri`, it passes `$RemainingArguments` as a named parameter in $PSBoundParameters collection!

There are a couple of ways to fix this (in my order of preference):

1. Remove the `RemainingArguments` parameter from $PSBoundParameters before calling Resolve-AzDoModuleUri e.g. add `$null = $PSBoundParameters.Remove('RemainingArguments')` to line 43 in `Invoke-LabModuleDownloadFromAzDo`
3. Add the FeedCredential parameter to Resolve-AzDoModuleUri